File: recipiece-server/Database/Models/ShoppingList.py
from typing import TypedDict, List, Optional

from Database.Models import BaseModel


class ShoppingListItem(TypedDict, total=False):
    id: str
    name: str
    notes: str
    category: str
    ordinal: int
    completed: bool
    recipe: Optional[str]


class ShoppingList(BaseModel.UserOwnedModel, total=False):
    name: str
    listItems: List[ShoppingListItem]

File: recipiece-ui/src/app/api/model/shopping-list.ts
import {Model} from './model';

export interface IShoppingListItem {
  id: string;
  name: string;
  notes: string;
  category: string;
  ordinal: number;
  completed: boolean;
  recipe?: string;
}

export interface IShoppingList extends Model {
  _id?: string;
  name: string;
  listItems: IShoppingListItem[];
}
